app.py

```diff
@@ -1,113 +1,93 @@
 
-    # Trying to acquire token silently
-    if accounts:
-        result = application.acquire_token_silent(app_config.SCOPE, account=accounts[0])
-    return result
-
-
-def get_graph_info(result):
-    if 'access_token' not in result:
-        return flask.redirect(flask.url_for('index'))
-    endpoint = 'https://graph.microsoft.com/v1.0/me/'
-    http_headers = {'Authorization': 'Bearer ' + result['access_token'],
-                    'User-Agent': 'msal-python-sample',
-                    'Accept': 'application/json',
-                    'Content-Type': 'application/json',
-                    'client-request-id': str(uuid.uuid4())}
-    graph_data = requests.get(endpoint, headers=http_headers, stream=False).json()
-    return graph_data
+app = Flask(__name__)
+app.config["SESSION_TYPE"] = "filesystem"  # We choose to store tokens in server-side session
+Session(app)
 
 
-@app.route('/')
+@app.route("/")
 def index():
-    return render_template("index.html")
-
-
-@app.route('/processing')
-def processing():
-    # Initializing
-    is_session = session.get(request.cookies.get("session"))
-    if is_session is None:
-        session[request.cookies.get("session")] = ''
-    cache.deserialize(session.get(request.cookies.get("session")))
-    return flask.redirect(flask.url_for('my_info'))
-
-
-@app.route('/my_info')
-def my_info():
-    result = check_cache()
-    if result:
-        graph_result = get_graph_info(result)
-        return flask.render_template('display.html', auth_result=graph_result, cond="logout")
-    else:
-        return flask.render_template('display.html', auth_result="You are not signed in", cond="")
-
-
-@app.route('/authenticate')
-def authenticate():
-    # Call to the authorize endpoint
-    auth_state = str(uuid.uuid4())
-    session[(request.cookies.get("session")+'state')] = auth_state
-    authorization_url = application.get_authorization_request_url(app_config.SCOPE, state=auth_state,
-                                                                  redirect_uri=app_config.REDIRECT_URI)
-    resp = flask.Response(status=307)
-    resp.headers['location'] = authorization_url
-    return resp
-
-
-@app.route("/getAToken")
-def main_logic():
-    code = flask.request.args['code']
-    state = flask.request.args['state']
-    # Raising error if state does not match
-    if state != session[(request.cookies.get("session")+'state')]:
-        raise ValueError("State does not match")
-    result = application.acquire_token_by_authorization_code(code, scopes=app_config.SCOPE,
-                                                             redirect_uri=app_config.REDIRECT_URI)
-    # Updating cache
-    set_cache()
-
-    # Using access token from result to call Microsoft Graph
-    graph_data = get_graph_info(result)
-    return flask.render_template('display.html', auth_result=graph_data, cond="logout")
-
+    if not session.get("user"):
+        return redirect(url_for("login"))
+    return """Welcome, %s.
```

Comment:
Should we just keep the index.html instead of deleting it. I just believe it is much cleaner than having HTML in the application. Thoughts ?

Reply:
That's fair. Then perhaps you would also like a dedicated login.html too?